ocrd/ocrd/cli/workspace.py

```diff
@@ -151,10 +162,115 @@ def workspace_add_file(ctx, file_grp, file_id, mimetype, page_id, force, fname):
+    --file-grp by referencing the named group 'grp' in the regex as '{{ grp }}'.
+
+    \b
+    Example:
+        ocrd workspace bulk-add \\
+                --regex '^.*/(?P<fileGrp>[^/]+)/page_(?P<pageid>.*)\.(?P<ext>[^\.]*)$' \\
+                --file-id 'FILE_{{ fileGrp }}_{{ pageid }}' \\
+                --page-id 'PHYS_{{ pageid }}' \\
+                --file-grp "{{ fileGrp }}" \\
+                --url '{{ fileGrp }}/FILE_{{ pageid }}.{{ ext }}' \\
+                path/to/files/*/*.*
+
+    """
+    log = getLogger('ocrd.cli.workspace.bulk-add') # pylint: disable=redefined-outer-name
+    workspace = Workspace(ctx.resolver, directory=ctx.directory, mets_basename=ctx.mets_basename, automatic_backup=ctx.automatic_backup)
+
+    try:
+        pat = re.compile(regex)
+    except Exception as e:
+        log.error("Invalid regex: %s" % e)
+        sys.exit(1)
+
+    file_paths = []
+    for fglob in file_glob:
+        file_paths += [Path(x).resolve() for x in glob(fglob)]
+
+    _groups_seen = set()
+
+    for i, file_path in enumerate(file_paths):
+        log.info("[%4d/%d] %s" % (i, len(file_paths), file_path))
+
+        # match regex
+        m = pat.match(str(file_path))
+        if not m:
+            log.error("File not matched by regex: '%s'" % file_path)
+            sys.exit(1)
+        group_dict = m.groupdict()
+
+        # set up file info
+        file_dict = {'url': url, 'mimetype': mimetype, 'ID': file_id, 'pageId': page_id, 'fileGrp': file_grp}
+
+        # guess mime type
+        if not file_dict['mimetype']:
+            try:
+                file_dict['mimetype'] = EXT_TO_MIME[file_path.suffix]
+            except KeyError:
+                log.error("Cannot guess mimetype from extension '%s' for '%s'. Set --mimetype explicitly" % (file_path.suffix, file_path))
+                sys.exit(1)
+
+        # expand templates
+        for param_name in file_dict:
+            for group_name in group_dict:
+                file_dict[param_name] = file_dict[param_name].replace('{{ %s }}' % group_name, group_dict[group_name])
+
+        # copy files
+        if file_dict['url']:
+            urlpath = Path(workspace.directory, file_dict['url'])
+            if not dry_run and not urlpath.exists():
+                if not urlpath.parent.is_dir():
+                    urlpath.parent.mkdir()
+                urlpath.write_bytes(file_path.read_bytes())
+
+        # Honor --overwrite
+        fileGrp = file_dict.pop('fileGrp')
+        if not dry_run and overwrite and fileGrp not in _groups_seen:
+            workspace.remove_file_group(fileGrp, recursive=True, force=True, keep_files=False)
+            _groups_seen.add(fileGrp)
+
+        # Add to workspace (or not)
+        if dry_run:
+            log.info('workspace.add_file(%s)' % file_dict)
+        else:
+            workspace.add_file(fileGrp, ignore=ignore, force=force, **file_dict)
+
+    # save changes to disk
+    workspace.save_mets()
```

Comment:
This makes me think of another problem: what if at some point bulk-add exits with non-zero status, after already having copied a lot of files to their new location? Then the next time I try, these files already exist in the filesystem, perhaps getting in the way. (It's covered with the proposed `--skip` extension, but maybe there's a better solution which makes METS and FS actions atomic?)